python/ray/dataframe/dataframe.py

```diff
@@ -1291,9 +1331,30 @@ def describe_helper(df):
         return result
 
     def diff(self, periods=1, axis=0):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        """Finds the difference between elements on the axis requested
+
+        Args:
+            periods: Periods to shift for forming difference
+            axis: Take difference over rows or columns
+
+        Returns:
+            DataFrame with the diff applied
+        """
+
+        if (axis == 1 or axis == 'columns'):
+            result = _map_partitions(lambda df:
```

Comment:
Some code can be reused between the two functions here. There are examples that do similar things, eg: `mode`.